commands.txt

```diff
@@ -1,4 +1,13 @@
 if [ `uname` = 'Darwin' ]; then export SD_CURL_CMD_WRAPPER="eval"; else export SD_CURL_CMD_WRAPPER="$(echo 'sd-step exec --pkg-version 7.54.1 core/curl')"; fi
 cd $SD_ARTIFACTS_DIR
 find . -type f -print > manifest.txt
-find . -type f -exec sh -c 'f=`echo $1 | sed "s/^\.\///"`; encoded=$($SD_CURL_CMD_WRAPPER "curl -s -w \"%{url_effective}\" --data-urlencode \"$f\" -G \"\" | sed -e \"s/^\/?//\""); $SD_CURL_CMD_WRAPPER "curl -H \"Authorization: Bearer $SD_TOKEN\" -H \"Content-Type: text/plain\" -X PUT $STORE_URL/v1/builds/$SD_BUILD_ID/$ARTIFACTS_DIR_SUFFIX/$encoded -T ./\"$f\""' -- {} \;
+
+if [ "$SD_ZIP_ARTIFACTS" = "true" ]; then
+    ZIPFILE=${SD_BUILD_ID}_artifacts.zip
+    find . -type f -exec zip $ZIPFILE {} +
```

Comment:
do we have zip in all images ?

Reply:
Not sure. We can use `tar` instead

Reply:
Uhh, looks like tar can't create zip files, and lambda seems to be using ZipInputStream https://git.ouroath.com/cocktails-screwdriver/s3-lambda/blob/3e06565b16dcd7a0431458cf40b5f178c3f82751/src/main/java/com/oath/aws/screwdriver/lambda/S3EventProcessor.java#L44 So we might need the `zip` command

Reply:
you will need to run this in a hab step. See `SD_CURL_CMD_WRAPPER`